drivers/soundwire/intel.c

```diff
@@ -1129,11 +1129,24 @@ static void *intel_get_sdw_stream(struct snd_soc_dai *dai,
 	return dma->stream;
 }
 
+/* handle SUSPEND trigger to free the DAI widget during suspend */
+static int intel_trigger(struct snd_pcm_substream *substream, int cmd, struct snd_soc_dai *dai)
+{
+	struct sdw_cdns *cdns = snd_soc_dai_get_drvdata(dai);
+	struct sdw_intel *sdw = cdns_to_intel(cdns);
+
+	if (cmd == SNDRV_PCM_TRIGGER_SUSPEND)
+		return intel_free_stream(sdw, substream, dai, sdw->instance);
+
+	return 0;
+}
+
 static const struct snd_soc_dai_ops intel_pcm_dai_ops = {
 	.startup = intel_startup,
 	.hw_params = intel_hw_params,
 	.prepare = intel_prepare,
 	.hw_free = intel_hw_free,
+	.trigger = intel_trigger,
```

Comment:
we move trigger support to sof_sdw.c in machine driver since Linux sdw maintainer required ?